godot-macros/src/class/godot_api.rs

```diff
@@ -229,11 +232,29 @@ fn process_godot_fns(decl: &mut Impl) -> Result<(Vec<FuncDefinition>, Vec<Functi
                 return attr.bail("generic fn parameters are not supported", method);
             }
 
-            match attr.ty {
-                BoundAttrType::Func { rename } => {
+            match &attr.ty {
+                BoundAttrType::Func {
+                    rename,
+                    has_gd_self,
+                } => {
                     // Signatures are the same thing without body
-                    let sig = util::reduce_to_signature(method);
-                    func_definitions.push(FuncDefinition { func: sig, rename });
+                    let mut sig = util::reduce_to_signature(method);
+                    if *has_gd_self {
+                        if sig.params.is_empty() {
+                            return attr.bail("When the attribute gd_self is used the method must have a first argument whose type is Gd<Self>", method);
```

Comment:
Punctuation, and clarity of parameter vs. argument, and attribute key vs. attribute:
```suggestion
                            return attr.bail("with attribute key `gd_self`, the method must have a first parameter of type Gd<Self>", method);
```

Reply:
Yeah, there's a bunch of proper nouns that I don't know, definitely feel free to correct any of this.